### PR TITLE
Feature/star wars demo

### PR DESCRIPTION
**Pull request by bwalsh**
This PR:
* cleans checkin of star wars demo  (original PR #15 )
* rebased from main

Key changes from original PR:

* clarifies use of `templatePointers` object
* adds `reference.yaml` to type `{id:_}` style references
* adds  `tests/unit/test_swapi.py` to validate schema and instances
* leverages `VertexLinkWriter` to process instances and schema to create new instances with links
* bundles schema to `tests/fixtures/swapi/graph-swapi.json`

### CODE REVIEW

File: tests/fixtures/swapi/regenObjects.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        for target in ['from', 'to']:
+            loc = j[target].split(':')
+            j[loc[0].lower()] = loc[1]
+            if loc[0].lower() not in [x['rel'] for x in schemas[j['label']]['links']]:
+                schemas[j['label']]['links'].append({
+                    'rel': loc[0].lower(),
+                    'href': loc[0].lower() + '/{id}',
+                    'templateRequired': ['id'],
+                    'targetSchema': {'$ref': loc[0].lower() + '.yaml'},
+                    'templatePointers': 'id',
```
Comment (evergreen700) [inline tests/fixtures/swapi/regenObjects.py:106]:
```suggestion
                    'templatePointers': {'id': '/%s/-/id' % loc[0]},
```

File: tests/fixtures/swapi/regenObjects.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            schemas[j['label']]['properties'][k]['type'].append(typeStr(j[k]))
+
+for r in multTable.iter_rows(named=True):
+    targetType = r['to'].split(':')[0]
+    schemas[r['sourceType']]['links'].append({
+        'rel': r['label'].lower(),
+        'href': targetType.lower() + '/{id}',
+        'templateRequired': ['id'],
+        'targetSchema': {'$ref': targetType.lower() + '.yaml'},
+        'templatePointers': '/id',
```
Comment (evergreen700) [inline tests/fixtures/swapi/regenObjects.py:132]:
```suggestion
        'templatePointers': {'id': '/%s/-/id' % loc[0]},
```

File: tests/fixtures/swapi/regenObjects.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        'targetHints': {
+            'directionality': 'outbound',
+            'multiplicity': 'has_many' if r["toMultiple"] else 'has_many',
+            'backref': r['sourceType'].lower() + ('s' if r['fromMultiple'] and r['sourceType'][-1] != 's' else "")
+        }
+    })
```
Comment (evergreen700) [inline tests/fixtures/swapi/regenObjects.py:138]:
```suggestion
    })
    schemas[r['sourceType']]['properties'][r['label']] = {'type': 'array', 'items': {'$ref': 'reference.yaml'}} 
```

**Review by evergreen700 — APPROVED** [review]
Changes to make to the generation script are suggested but overall the schemas look good. I am a bit confused why 'homeworld' is the only property you changed to be stored as a single object rather than an object array with only one element (as seen with other to-one properties like 'species') but at the very least each property is consistent with itself and that's all we need to use this as a test case.

**Review by bwalsh — COMMENTED** [review]
LGTM, lets commit and open a new PR for future changes

**Pull request merged**